phylo2vec/src/tree_vec/ops/newick/mod.rs

```diff
@@ -50,6 +52,79 @@ fn _get_cherries_recursive_inner(ancestry: &mut Ancestry, newick: &str, has_pare
     }
 }
 
+fn stoi_substr(s: &str, start: usize, end: &mut usize) -> Result<usize, IntErrorKind> {
+    let s = &s[start..];
+    let mut value = 0;
+    for (i, c) in s.chars().enumerate() {
+        if let Some(digit) = c.to_digit(10) {
+            value = value * 10 + digit as usize;
+            *end = start + i + 1;
+        } else {
+            break;
+        }
+    }
+    if *end > start {
+        Ok(value)
+    } else {
+        Err(std::num::IntErrorKind::Empty)
+    }
+}
+
+fn _get_cherries(ancestry: &mut Ancestry, newick: &str) {
```

Comment:
```suggestion
fn _get_cherries_inner(ancestry: &mut Ancestry, newick: &str) {
```
Suggesting name change for readability so as to differentiate between this function and the public `get_cherries` in this file